### ISSUE TITLE
libfido2.so missing symbols fido_dev_new_with_info and fido_dev_open_with_info

### ISSUE THREAD

**Issue by floppym** [Yubico/libfido2#389]
**What version of libfido2 are you using?**
git master (4d45fc92bde4766874168cf412eac21bc059c0d8)
Also happens with 1.7.0 and 1.8.0.
**What operating system are you running?**
Gentoo Linux
**How does the problem manifest itself?**
Build failure
**Is the problem reproducible?**
Yes
**What are the steps that lead to the problem?**
Run `cmake -DBUILD_STATIC_LIBS=OFF`, and then attempt to build.
```
FAILED: examples/cred 
: && /usr/bin/cc -D_POSIX_C_SOURCE=200809L -D_BSD_SOURCE -D_GNU_SOURCE -D_DEFAULT_SOURCE -std=c99 -Wl,-z,noexecstack -Wl,-z,relro,-z,now   -rdynamic  -fPIE -pie examples/CMakeFiles/cred.dir/cred.c.o examples/CMakeFiles/cred.dir/util.c.o examples/CMakeFiles/cred.dir/__/openbsd-compat/clock_gettime.c.o examples/CMakeFiles/cred.dir/__/openbsd-compat/getopt_long.c.o examples/CMakeFiles/cred.dir/__/openbsd-compat/strlcat.c.o examples/CMakeFiles/cred.dir/__/openbsd-compat/strlcpy.c.o -o examples/cred  -Wl,-rpath,/home/floppym/src/libfido2/build/src  src/libfido2.so.1.9.0  -lcbor  -lcrypto  -ludev  -lz && :
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: examples/CMakeFiles/cred.dir/cred.c.o: in function `open_from_manifest':
cred.c:(.text+0x5d7): undefined reference to `fido_dev_new_with_info'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: cred.c:(.text+0x5e7): undefined reference to `fido_dev_open_with_info'
collect2: error: ld returned 1 exit status
```

**Comment by martelletto** [comment]
Thank you for the report. Due to historical reasons, these functions are not part of the official API. We will look into either making them official or dropping them from the set of supplied examples. In the meantime, please specify `-DBUILD_EXAMPLES=OFF` if using `-DBUILD_STATIC_LIBS=OFF`.

**Comment by floppym** [comment]
Ah, I see that the examples don't get installed anyway, so I disabled them unconditionally in our build script.

**Comment by martelletto** [comment]
Addressed in bf2e351fc6.
